src/tools/cephfs/top/cephfs-top

```diff
@@ -573,22 +662,77 @@ class FSTop(object):
+                self.set_option(c)
             except curses.error:
                 # this may happen when addstr the terminal window
                 # size changed, just retry it
                 pass
 
+    def run_all_display(self, fs_l):
+        while not self.exit_ev.is_set():
+            # use stdscr.clear() instead of clearing each window
+            # to avoid screen blinking.
+            self.stdscr.clear()
+            self.refresh_window_size()
+            if self.height <= 4 or self.width <= 2:
+                self.exit_ev.wait(timeout=self.refresh_interval_secs)
+                continue
+
+            # coordinate constants for windowing -- (height, width, y, x)
+            # NOTE: requires initscr() call before accessing COLS, LINES.
+            try:
+                HEADER_WINDOW_COORD = (3, self.width - 1, 0, 0)
+                self.header = curses.newwin(*HEADER_WINDOW_COORD)
+            except curses.error:
+                # this may happen when creating the sub windows the
+                # terminal window size changed, just retry it
+                pass
+
+            stats_json = self.perf_stats_query()
+            try:
+                main, top, num_client = 6, 3, 0
+                for fs in fs_l:
+                    if self.height >= 4 + num_client:
+                        TOPLINE_WINDOW_COORD = (3, self.width - 1, top, 0)
+                        self.topl = curses.newwin(*TOPLINE_WINDOW_COORD)
+                    else:
+                        self.topl = None
+                    if self.height >= 7 + num_client:
+                        MAIN_WINDOW_COORD = (self.height - 6, self.width - 1,
+                                             main, 0)
+                        self.mainw = curses.newwin(*MAIN_WINDOW_COORD)
+                    else:
+                        self.mainw = None
+
+                    num_client = len(stats_json[CLIENT_METADATA_KEY][fs])
+                    top = top + (num_client + 4)
+                    main = main + (num_client + 4)
+                    if self.refresh_header(stats_json, fs):
+                        x_coord_map = self.refresh_top_line_and_build_coord()
+                        self.refresh_main_window(x_coord_map, stats_json, fs)
+                self.exit_ev.wait(timeout=self.refresh_interval_secs)
+                curses.halfdelay(self.refresh_interval_secs)
+                c = self.header.getch()
+                self.set_option(c)
+            except curses.error:
+                # this may happen when addstr the terminal window
+                # size changed, just retry it
+                continue
+
 
 if __name__ == '__main__':
     def float_greater_than(x):
         value = float(x)
         if value < MIN_REFRESH_INTERVAL:
             raise argparse.ArgumentTypeError(
-                f'Refresh interval should be greater than or equal to {MIN_REFRESH_INTERVAL}')
+                'Refresh interval should be greater than or equal to'
```

Comment:
unrelated change?

Reply:
This was done to remove some flake8 errors.

Reply:
have this as a separate commit.